client/asset/btc/btc.go

```diff
@@ -1385,6 +1387,96 @@ func (btc *baseWallet) Info() *asset.WalletInfo {
 	return btc.walletInfo
 }
 
+func (btc *baseWallet) startTxHistoryDB() error {
+	dir, err := os.Open(btc.walletDir)
+	if err != nil {
+		return fmt.Errorf("error opening wallet directory: %w", err)
+	}
+	defer dir.Close()
+
+	entries, err := dir.Readdir(0)
+	if err != nil {
+		return fmt.Errorf("error reading wallet directory: %w", err)
+	}
+
+	pattern := regexp.MustCompile(`^txhistory-(.+)\.db$`)
+
+	var dbPath string
+	for _, entry := range entries {
+		if !entry.IsDir() {
+			continue
+		}
+
+		match := pattern.FindStringSubmatch(entry.Name())
+		if match == nil {
+			continue
+		}
+
+		address := match[1]
+		owns, err := btc.OwnsDepositAddress(address)
+		if err != nil {
+			btc.log.Errorf("Error checking if wallet owns deposit address %s: %w", address, err)
+			continue
+		}
+		if owns {
+			btc.log.Infof("Using tx history db %s", entry.Name())
+			dbPath = filepath.Join(btc.walletDir, entry.Name())
+			break
+		}
+	}
+
+	if dbPath == "" {
+		depositAddr, err := btc.DepositAddress()
```

Comment:
Ideally we wouldn't generate a new address just for this. Decred's spv wallet has `CurrentAddress`. There's also `AddressAtIdx`, which would be an interesting solution to the problem described above too. Unfortunately `AddressAtIdx` is only exposed through the grpc api, so it doesn't adapt to rpc.

Reply:
I'll try to use `getaddressesbylabel` in btc and `getaddressesbyaccount` and dcr.